examples/model_selection/plot_precision_recall.py

```diff
@@ -71,76 +85,148 @@
              :func:`sklearn.metrics.precision_score`,
              :func:`sklearn.metrics.f1_score`
 """
-print(__doc__)
-
-import matplotlib.pyplot as plt
-import numpy as np
-from itertools import cycle
+from __future__ import print_function
 
+###############################################################################
+# In binary classification settings
+# --------------------------------------------------------
+#
+# Create simple data
+# ..................
+#
+# Try to differentiate the two first classes of the iris data
 from sklearn import svm, datasets
-from sklearn.metrics import precision_recall_curve
-from sklearn.metrics import average_precision_score
 from sklearn.model_selection import train_test_split
-from sklearn.preprocessing import label_binarize
-from sklearn.multiclass import OneVsRestClassifier
+import numpy as np
 
-# import some data to play with
 iris = datasets.load_iris()
 X = iris.data
 y = iris.target
 
-# setup plot details
-colors = cycle(['navy', 'turquoise', 'darkorange', 'cornflowerblue', 'teal'])
-lw = 2
-
-# Binarize the output
-y = label_binarize(y, classes=[0, 1, 2])
-n_classes = y.shape[1]
-
 # Add noisy features
 random_state = np.random.RandomState(0)
 n_samples, n_features = X.shape
 X = np.c_[X, random_state.randn(n_samples, 200 * n_features)]
 
+# Limit to the two first classes, and split into training and test
+X_train, X_test, y_train, y_test = train_test_split(X[y < 2], y[y < 2],
+                                                    test_size=.5,
+                                                    random_state=random_state)
+
+# Create a simple classifier
+classifier = svm.SVC(kernel='linear', probability=True,
```

Comment:
Why????? How about ``LogisticRegression``? Or not using ``probability=True``? We don't need probabilities for the precision-recall-curve and many people are not aware of that. So maybe using ``SVC(linear)`` is actually good, but we shouldn't use ``probability=True``. What's the reason not to use ``LinearSVC()``?